src/Plant.py

```diff
@@ -19,9 +21,7 @@ def setWatering_time(self,new_time):
     def setLast_watered(self,new_watering):
         self.__last_watered = new_watering
 
-    #When obj is initialized, this sets a name, period between watering, what time the plant is watered, and when the plant was last watered
-    def __init__(self,plant_name, plant_watering_interval, plant_watering_time, plant_last_watered):
-        self.setName(plant_name)
+    #When obj is initialized a period between watering and when the plant was last watered
+    def __init__(self,plant_watering_interval = datetime.timedelta(1), plant_last_watered = datetime.datetime.now()):
```

Comment:
What's your rationale for the changes you opted for here? (Aside from the obvious "you wanted me to revisit them." 😆  ) ie - why did you opt to remove the ones you did and keep the ones you kept?

Reply:
Keeping the parameters that I did was important in order to calculate when the plant needed watered. Without the time interval, there would be no way to calculate the next time a plant needed watered without asking for the interval again. The same story applies for the time it was last watered. 
The watering time was trivial because I intended to use it for no other purpose than to remind the user at a specific time of the day to water the plant. Similar to watering time, name was unimportant because of its disconnect with the logic behind watering and it would rather be an organizational thing.  
If I may ask, what would be a better name for last_watered? It is supposed to simply act as the date as to which the interval will be applied in order to calculate when the plant needs water. In typical use, it will simply be the previous date it was watered, but something just doesn't sit right with this.

Reply:
> name was unimportant because of its disconnect with the logic behind watering and it would rather be an organizational thing.

Okay, so do you have something in mind for how to identify the Plant in question?

> If I may ask, what would be a better name for last_watered? It is supposed to simply act as the date as to which the interval will be applied in order to calculate when the plant needs water. In typical use, it will simply be the previous date it was watered, but something just doesn't sit right with this.

First, this part:

> but something just doesn't sit right with this

is a fantastic self-observation! Always honor this gut instinct, because it's very often an indicator of a "code smell" or that you're on the cusp of new information (basically, you know enough to know _that_ something is wrong, even if you can't yet pinpoint _what_ is wrong).

Now, as for the matter at hand... 🤔 

They say one of the hardest problems in computing is naming things, and this is why! 😅 

I don't have a good alternative right now, mainly because "last watered" is adequately descriptive and succinct. You could probably drop the `plant_` prefixes, though. You're working in the context of a plant, and any work with those outside of this class will already be labeled as a plant via the naming, making the prefix redundant.

If it bothers you enough, add a `TODO:` note in the comments above it and write out your intention for it and that it doesn't feel right, but you don't know what would work better there. You can always revisit it later. For now, don't get _too_ hung up on it. Don't let perfect be the enemy of good.

Reply:
I added the 'plant_' prefixes because I intended to define some attributes with the same name and I didn't want to have to deal with accidentally swapping the obj attribute and the parameter in my code (if I am thinking correctly then `self.x` and `x` are two different variables in this situation). Also, when I get to beginning to actually implement this class in a program, a name should still be able to be defined with the `setName()` function and returned with the `getName()` function. I just don't consider it absolutely necessary for the logic functions of the object.

Also, I do realize that the attributes are not defined correctly in the code, I have to do some research on that before I get to it.